File: VBHotelDbLib/Model/Facilities.vb
Namespace Model
    Public Class Facilities
        Private _faciId As Integer
        Private _faciName As String
        Private _faciDescription As String
        Private _faciMaxNumber As Integer
        Private _faciMeasureUnit As String
        Private _faciRoomNumber As String
        Private _faciStartdate As DateTime
        Private _faciEndate As DateTime
        Private _faciLowPrice As Double
        Private _faciHighPrice As Double
        Private _faciRatePrice As Double
        Private _faciDiscount As Single
        Private _faciTaxRate As Single
        Private _faciModifiedDate As DateTime
        Private _faciCagroId As Integer
        Private _faciHotelId As Integer

        Public Sub New()
        End Sub

        Public Sub New(faciId As Integer, faciName As String, faciDescription As String, faciMaxNumber As Integer, faciMeasureUnit As String, faciRoomNumber As String, faciStartdate As Date, faciEndate As Date, faciLowPrice As Double, faciHighPrice As Double, faciRatePrice As Double, faciDiscount As Single, faciTaxRate As Single, faciModifiedDate As Date, faciCagroId As Integer, faciHotelId As Integer)
            Me.FaciId = faciId
            Me.FaciName = faciName
            Me.FaciDescription = faciDescription
            Me.FaciMaxNumber = faciMaxNumber
            Me.FaciMeasureUnit = faciMeasureUnit
            Me.FaciRoomNumber = faciRoomNumber
            Me.FaciStartdate = faciStartdate
            Me.FaciEndate = faciEndate
            Me.FaciLowPrice = faciLowPrice
            Me.FaciHighPrice = faciHighPrice
            Me.FaciRatePrice = faciRatePrice
            Me.FaciDiscount = faciDiscount
            Me.FaciTaxRate = faciTaxRate
            Me.FaciModifiedDate = faciModifiedDate
            Me.FaciCagroId = faciCagroId
            Me.FaciHotelId = faciHotelId
        End Sub

        Public Property FaciId As Integer
            Get
                Return _faciId
            End Get
            Set(value As Integer)
                _faciId = value
            End Set
        End Property

        Public Property FaciName As String
            Get
                Return _faciName
            End Get
            Set(value As String)
                _faciName = value
            End Set
        End Property

        Public Property FaciDescription As String
            Get
                Return _faciDescription
            End Get
            Set(value As String)
                _faciDescription = value
            End Set
        End Property

        Public Property FaciMaxNumber As Integer
            Get
                Return _faciMaxNumber
            End Get
            Set(value As Integer)
                _faciMaxNumber = value
            End Set
        End Property

        Public Property FaciMeasureUnit As String
            Get
                Return _faciMeasureUnit
            End Get
            Set(value As String)
                _faciMeasureUnit = value
            End Set
        End Property

        Public Property FaciRoomNumber As String
            Get
                Return _faciRoomNumber
            End Get
            Set(value As String)
                _faciRoomNumber = value
            End Set
        End Property

        Public Property FaciStartdate As Date
            Get
                Return _faciStartdate
            End Get
            Set(value As Date)
                _faciStartdate = value
            End Set
        End Property

        Public Property FaciEndate As Date
            Get
                Return _faciEndate
            End Get
            Set(value As Date)
                _faciEndate = value
            End Set
        End Property

        Public Property FaciLowPrice As Double
            Get
                Return _faciLowPrice
            End Get
            Set(value As Double)
                _faciLowPrice = value
            End Set
        End Property

        Public Property FaciHighPrice As Double
            Get
                Return _faciHighPrice
            End Get
            Set(value As Double)
                _faciHighPrice = value
            End Set
        End Property

        Public Property FaciRatePrice As Double
            Get
                Return _faciRatePrice
            End Get
            Set(value As Double)
                _faciRatePrice = value
            End Set
        End Property

        Public Property FaciDiscount As Single
            Get
                Return _faciDiscount
            End Get
            Set(value As Single)
                _faciDiscount = value
            End Set
        End Property

        Public Property FaciTaxRate As Single
            Get
                Return _faciTaxRate
            End Get
            Set(value As Single)
                _faciTaxRate = value
            End Set
        End Property

        Public Property FaciModifiedDate As Date
            Get
                Return _faciModifiedDate
            End Get
            Set(value As Date)
                _faciModifiedDate = value
            End Set
        End Property

        Public Property FaciCagroId As Integer
            Get
                Return _faciCagroId
            End Get
            Set(value As Integer)
                _faciCagroId = value
            End Set
        End Property

        Public Property FaciHotelId As Integer
            Get
                Return _faciHotelId
            End Get
            Set(value As Integer)
                _faciHotelId = value
            End Set
        End Property

        Public Overrides Function ToString() As String
            Return $"
Fasilitas ID : {FaciId}{vbTab} | Fasilitas : {FaciName}	
Deskripsi : {FaciDescription}	
Berisi : {FaciMeasureUnit}     {vbTab} | Kapasitas : {FaciMaxNumber} {vbTab} {vbTab} | Number : {FaciRoomNumber}	
Harga Rendah : {FaciLowPrice} {vbTab} | Harga Maksimum : {FaciHighPrice} {vbTab} | Pajak  : {FaciTaxRate}
Nomor Kategori : {FaciCagroId}  {vbTab} | HotelId : {FaciHotelId} {vbTab}
Tanggal Awal {vbTab}{vbTab} : {FaciStartdate} 
Tanggal Akhir {vbTab}{vbTab} : {FaciEndate}
Tanggal Pembaharuan {vbTab} : {FaciModifiedDate}
"
        End Function
    End Class
End Namespace

File: RealtaClient/HandleConsole/Facilities.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;
using VBHotelDbLib.HotelVbApi;

namespace HotelConsole.HandleConsole
{
    internal class Facilities
    {
        public static void Run(IHotelDbLib config)
        {
            Facilities res = new Facilities();

            res.GetAllDataFaci(config);
            res.GetDataFaciById(config, 27);
            res.CreateFaci(config);
            res.UpdateFaci(config);
            res.UpdateFaciBySP(config);
            res.DeleteFaci(config, 28);

            Task s = res.GetDataFaciAsync(config);
            Thread.Sleep(25);
        }

        private void GetAllDataFaci(IHotelDbLib config)
        {
            var getAllDataFaci = config.RepositoryManager.Facilities.FindAllFacilities();
            foreach (var x in getAllDataFaci) Console.WriteLine(x);
        }

        private void GetDataFaciById(IHotelDbLib config, int id)
        {
            var facilitiesById = config.RepositoryManager.Facilities.FindFacilitiesById(id);
            Console.WriteLine($"RESULT : " +
                              $"{facilitiesById}");
        }

        private void CreateFaci(IHotelDbLib config)
        {
            var newFacilities = config.RepositoryManager.Facilities.CreateFacilities(new VBHotelDbLib.Model.Facilities
            {
                FaciName = "Test New Facilities",
                FaciDescription = null,
                FaciMaxNumber = 0,
                FaciMeasureUnit = null,
                FaciRoomNumber = "TEST04",
                FaciStartdate = new DateTime(2023, 1, 1, 6, 51, 39),
                FaciEndate = new DateTime(2024, 1, 1, 6, 51, 39),
                FaciLowPrice = 200000,
                FaciHighPrice = 250000,
                FaciRatePrice = 230000,
                FaciDiscount = 25000,
                FaciTaxRate = 10000,
                FaciModifiedDate = DateTime.Now,
                FaciCagroId = 1,
                FaciHotelId = 4
            });

            Console.WriteLine($"New Hotel Review Created : {newFacilities}");
        }

        private void UpdateFaci(IHotelDbLib config)
        {
            //Data Update!
            var id = 27;
            var faciName = "Test New Facilities";
            var faciDescription = "Updated FacilitiesName";
            var faciMaxNumber = 0;
            //var faciMeasureUnit = null;
            var faciRoomNumber = "TEST04";
            var faciStartdate = new DateTime(2023, 1, 1, 6, 51, 39);
            var faciEndate = new DateTime(2024, 1, 1, 6, 51, 39);
            var faciLowPrice = 200000;
            var faciHighPrice = 250000;
            var faciRatePrice = 230000;
            var faciDiscount = 25000;
            var faciTaxRate = 10000;
            var faciModifiedDate = new DateTime(2023,1,11,00,26,16);
            var faciCagroId =1;
            var faciHotelId =4;

            var updateFacilities = config.RepositoryManager.Facilities.UpdateFacilitiesById(id, faciName, faciDescription, faciMaxNumber, null, faciRoomNumber, faciStartdate, faciEndate, faciLowPrice, faciHighPrice, faciRatePrice, faciDiscount, faciTaxRate, faciModifiedDate, faciCagroId, faciHotelId);
            var facilitiesResult = config.RepositoryManager.Facilities.FindFacilitiesById(id);
            Console.WriteLine(facilitiesResult);
        }

        private void UpdateFaciBySP(IHotelDbLib config)
        {
            //Data Update!
            var id = 27;
            var faciName = "Test New Facilities";
            var faciDescription = "Updated FacilitiesName BySP";
            var faciMaxNumber = 0;
            var faciMeasureUnit = "people";
            var faciRoomNumber = "TEST04";
            var faciStartdate = new DateTime(2023, 1, 1, 6, 51, 39);
            var faciEndate = new DateTime(2024, 1, 1, 6, 51, 39);
            var faciLowPrice = 200000;
            var faciHighPrice = 250000;
            var faciRatePrice = 230000;
            var faciDiscount = 25000;
            var faciTaxRate = 10000;
            var faciModifiedDate = new DateTime(2023, 1, 11, 00, 26, 16);
            var faciCagroId = 1;
            var faciHotelId = 4;

            var updateFacilitiesBySp = config.RepositoryManager.Facilities.UpdateFacilitiesBySp(id, faciName, faciDescription, faciMaxNumber, faciMeasureUnit, faciRoomNumber, faciStartdate, faciEndate, faciLowPrice, faciHighPrice, faciRatePrice, faciDiscount, faciTaxRate, faciModifiedDate, faciCagroId, faciHotelId);
            var facilitiesResult = config.RepositoryManager.Facilities.FindFacilitiesById(id);
            Console.WriteLine(facilitiesResult);
        }

        private void DeleteFaci(IHotelDbLib config, int id)
        {
            var rowDelete = config.RepositoryManager.Facilities.DeleteFacilities(id);
            Console.WriteLine($"Delete Facilities Reviews row : {rowDelete}" +
                              $"");
            GetDataFaciById(config, id);
        }

        private async Task GetDataFaciAsync(IHotelDbLib config)
        {
            Console.WriteLine("============ini adalah asynchronous===========");

            var dataFacilities = await config.RepositoryManager.Facilities.FindAllFacilitiesAsync();
            foreach (var item in dataFacilities) Console.WriteLine(item);
        }
    }
}
